great_expectations/core/util.py

```diff
@@ -407,6 +407,7 @@ def datetime_to_int(dt: datetime.date) -> int:
 class AzureUrl:
     """
     Parses an Azure Blob Storage URL into its separate components
+    Format: <ACCOUNT_NAME>.blob.core.windows.net/<CONTAINER>/<BLOB>
```

Comment:
I've made a few Azure related changes here since I was already in the same files.

If you think I should move these to a separate PR, please let me know! Don't want to make reviewing more complicated than it needs to be

Reply:
It's ok. The changes are small and are pretty closely related with GCSUrl